Description: Chore: use espree.latestEcmaVersion in config-initializer
Origin: upstream, https://github.com/eslint/eslint/commit/e13256e
Bug: https://github.com/eslint/eslint/issues/13157
Author: Kai Cataldo <7041728+kaicataldo@users.noreply.github.com>
Last-Update: 2022-02-26
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/lib/init/config-initializer.js
+++ b/lib/init/config-initializer.js
@@ -15,6 +15,7 @@
     inquirer = require("inquirer"),
     ProgressBar = require("progress"),
     semver = require("semver"),
+    espree = require("espree"),
     recConfig = require("../../conf/eslint-recommended"),
     ConfigOps = require("../shared/config-ops"),
     log = require("../shared/logging"),
@@ -31,8 +32,6 @@
 // Private
 //------------------------------------------------------------------------------
 
-const DEFAULT_ECMA_VERSION = 2018;
-
 /* istanbul ignore next: hard to test fs function */
 /**
  * Create .eslintrc file in the current working directory
@@ -266,8 +265,7 @@
         extends: []
     };
 
-    // set the latest ECMAScript version
-    config.parserOptions.ecmaVersion = DEFAULT_ECMA_VERSION;
+    config.parserOptions.ecmaVersion = espree.latestEcmaVersion;
     config.env.es6 = true;
     config.globals = {
         Atomics: "readonly",
--- a/tests/lib/init/config-initializer.js
+++ b/tests/lib/init/config-initializer.js
@@ -15,6 +15,7 @@
     os = require("os"),
     sinon = require("sinon"),
     sh = require("shelljs"),
+    espree = require("espree"),
     autoconfig = require("../../../lib/init/autoconfig"),
     npmUtils = require("../../../lib/init/npm-utils");
 
@@ -137,7 +138,7 @@
                 assert.strictEqual(config.env.es6, true);
                 assert.strictEqual(config.globals.Atomics, "readonly");
                 assert.strictEqual(config.globals.SharedArrayBuffer, "readonly");
-                assert.strictEqual(config.parserOptions.ecmaVersion, 2018);
+                assert.strictEqual(config.parserOptions.ecmaVersion, espree.latestEcmaVersion);
                 assert.strictEqual(config.parserOptions.sourceType, "module");
                 assert.strictEqual(config.env.browser, true);
                 assert.strictEqual(config.extends, "eslint:recommended");
@@ -155,7 +156,7 @@
                 const config = init.processAnswers(answers);
 
                 assert.strictEqual(config.parserOptions.ecmaFeatures.jsx, true);
-                assert.strictEqual(config.parserOptions.ecmaVersion, 2018);
+                assert.strictEqual(config.parserOptions.ecmaVersion, espree.latestEcmaVersion);
                 assert.deepStrictEqual(config.plugins, ["react"]);
             });
 
@@ -163,7 +164,7 @@
                 answers.framework = "vue";
                 const config = init.processAnswers(answers);
 
-                assert.strictEqual(config.parserOptions.ecmaVersion, 2018);
+                assert.strictEqual(config.parserOptions.ecmaVersion, espree.latestEcmaVersion);
                 assert.deepStrictEqual(config.plugins, ["vue"]);
                 assert.deepStrictEqual(config.extends, ["eslint:recommended", "plugin:vue/essential"]);
             });
